.github/workflows/coverage.yml

```
@@ -41,4 +41,4 @@ jobs:
           pytest --cov=networkx --runslow --doctest-modules --durations=20 --pyargs networkx
 
       - name: Upload to codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
```

```suggestion
        uses: codecov/codecov-action@v3
```

There is some issues with moving to v4. I will look into them later.